lib.js

```diff
@@ -168,3 +171,32 @@ export function storeListResponseToString (res, opts = {}) {
     return res.results.map(({ link }) => link.toString()).join('\n')
   }
 }
+
+/**
+ * Return validated CARLink or undefined
+ * @param {AnyLink} cid
+ **/
+export function asCarLink (cid) {
+  if (cid.version === 1 && cid.code === CAR.codec.code) {
+    return /** @type {CARLink} */ (cid)
+  }
+}
```

Comment:
Alternatively you could make this a type assertion function:

```js
/**
 * Return true if CID is a CAR CID and false if not.
 * @param {import('multiformats').UnknownLink} cid
 * @returns {cid is import('multiformats').Link<Uint8Array, typeof CAR.codec.code>}
 */
export function isCarLink (cid) {
  return cid.version === 1 && cid.code === CAR.codec.code
}
```

...and then typescript knows it's a CAR CID later in the code and you don't have to re-assign e.g.:

<img width="431" alt="Screenshot 2023-10-02 at 16 06 14" src="https://github.com/web3-storage/w3cli/assets/152863/720b3909-df0e-4b1f-8776-1fc386f34dfc">